blocks/library/list/index.js

```diff
@@ -72,6 +112,17 @@ registerBlock( 'core/list', {
 		return (
 			<Editable
 				tagName={ nodeName.toLowerCase() }
+				onConfig={ ( settings ) => ( {
+					...settings,
+					plugins: ( settings.plugins || [] ).concat( 'lists' ),
+					lists_indent_on_tab: false,
```

Comment:
In normal TinyMCE you can escape the list by using the arrow keys and then tabbing works as normal. The problem is that because TinyMCE does not extend out of the list in this case you can't escape the list block and hence it is impossible to navigate away only using the keyboard.

It is a tricky problem, possibly another keybinding could be used for indenting or possibly the arrow keys could be used for inter-block navigation when you are on the first/last line.